tests/base/test_context.py

```diff
@@ -881,3 +884,137 @@ def test_check_allowlist_and_denylist(monkeypatch: MonkeyPatch):
 
     validate_channels(("defaults",))
     validate_channels((DEFAULT_CHANNELS[0], DEFAULT_CHANNELS[1]))
+
+
+def test_pkgs_envs_default_dirs(testdata, unset_condarc_pkgs):
+    """Test that the default envs/pkgs directories are not in the base environment."""
+    user_data = Path(user_data_dir(APP_NAME, APP_NAME, None, False))
+    envs = user_data / "envs"
+    pkgs = user_data / "pkgs"
+
+    # The pkgs and envs should exist in the user data dir
+    assert str(envs) in set(context.envs_dirs)
+    assert str(pkgs) in set(context.pkgs_dirs)
+
+    # They should not be in the prefix
+    prefix = Path(context.conda_prefix)
+    assert not envs.is_relative_to(prefix)
+    assert not pkgs.is_relative_to(prefix)
+
+
+def test_set_pkgs_envs_default_dirs(testdata, unset_condarc_pkgs):
+    """Test that the envs/pkgs directories revert to legacy behavior if set."""
+    envs = "/usr/local/foo/envs"
+    pkgs = "/usr/local/foo/pkgs"
+
+    additional_settings = dals(
+        f"""
+        envs_dirs:
+          - {envs}
+        pkgs_dirs:
+          - {pkgs}
+        """
+    )
+
+    # Set up the context
+    reset_context()
+    condarc = yaml_round_trip_load(TEST_CONDARC + additional_settings)
+    context._set_raw_data(
+        {"testdata": YamlRawParameter.make_raw_parameters("testdata", condarc)}
+    )
+
+    # Track the calls to the context methods that get the old prefix
+    # locations of the pkgs and envs
+    with (
+        mock.patch.object(
+            context, "_prefix_pkgs_dirs", wraps=context._prefix_pkgs_dirs
+        ) as mock_prefix_pkgs,
+        mock.patch.object(
+            context, "_prefix_envs_dirs", wraps=context._prefix_envs_dirs
+        ) as mock_prefix_envs,
+    ):
+        context_envs_dirs = context.envs_dirs
+        context_pkgs_dirs = context.pkgs_dirs
```

Comment:
`mocker.spy` is perfect for this:

```suggestion
    mock_prefix_pkgs = mocker.spy(context, "_prefix_pkgs_dirs")
    mock_prefix_envs = mocker.spy(context, "_prefix_envs_dirs")
    context_envs_dirs = context.envs_dirs
    context_pkgs_dirs = context.pkgs_dirs
```